src/foam/flow/Document.js

```diff
@@ -327,33 +336,42 @@ foam.CLASS({
 
         'code': function(code) {
           return function(x) {
-            this.
-              start('pre').
-              cssClass('foam-flow-Document-code').
-              add(code).
-              end();
+            this.start('code').add(code).end();
           };
         },
 
         'foam': function(attributes) {
           return function(x) {
-            // TODO: Reuse FoamTagLoader support, support classloading
             var viewName = attributes.view;
             var className = attributes.class;
-            var cls = x.lookup(className, true);
-            var view = x.lookup(viewName, true);
 
-            if ( className && ! cls )
-              this.add('Unknown class', className);
-            if ( viewName && ! view )
-              this.add('Unknown view', viewName);
+            var slot = foam.flow.PromiseSlot.create({
```

Comment:
As discussed, let's move PromiseSlot to core/Slot.js as its probably useful, but let's also update Element.js to support just adding a Promise to the DOM

Reply:
Moved to Slot.js and added a promise(promise, func) method to Element.